core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -261,7 +262,30 @@ protected void parse() throws ProtocolException {
         parseTransactions(offset + HEADER_SIZE);
         length = cursor - offset;
     }
-    
+
+    static Block createGenesis(NetworkParameters n) {
```

Comment:
Well, it was private in `NetworkParameters` and I'm not sure how broadly we want it be used, so I think that I why I made it package-private. In its current form it requires significant mutation of the `Block` before it can be used.

I'm happy to make it `public` though. When were done with the refactoring there's no reason why such a method shouldn't be.

Reply:
You're right. I'll leave it at package-private for now. We can open it up later when we have reason to.